### PR TITLE
backport #35574 (Fix wrong error message for user activation)

### PR DESCRIPTION
**Pull request by coolcat-creations**
This PR is to Backport #35574 - I hope it's done correctly

Pull Request for Issue #35849 .

### Summary of Changes
Copied the code from Joomla 4 and added Js ;-)

### Testing Instructions
Testing Instructions

1.     Set registration to user activation
2.     Register a new user
3.     click on the verification link
4.     click again on the verification link

Actual result BEFORE applying this Pull Request

grafik
Expected result AFTER applying this Pull Request

User is redirected to the login (with the same message as normal message)

### CODE REVIEW

File: components/com_users/controllers/registration.php [hunk 1]
```diff
@@ -63,7 +63,10 @@ public function activate()
 
 		if (!$userIdToActivate)
 		{
-			JError::raiseError(403, JText::_('COM_USERS_ACTIVATION_TOKEN_NOT_FOUND'));
+			$loginUrl = 'index.php?option=com_users&view=login';
```
Comment (Kubik-Rubik) [inline components/com_users/controllers/registration.php]: The introduction of the `$loginUrl` variable is unnecessary here. You can enter the URL directly in the `_` call.

Comment (coolcat-creations) [reply components/com_users/controllers/registration.php]: How do you mean?

File: components/com_users/controllers/registration.php [hunk 2]
```diff
@@ -63,7 +63,10 @@ public function activate()
...
+
+			$this->setMessage(JText::_('COM_USERS_ACTIVATION_TOKEN_NOT_FOUND'));
+			$this->setRedirect(JRoute::_($loginUrl, false));
```
Comment (Quy) [inline components/com_users/controllers/registration.php]:
```suggestion
			$this->setRedirect(JRoute::_('index.php?option=com_users&view=login', false));
```
Like this and remove line 66.

File: components/com_users/controllers/registration.php (hunk 1 above)
Comment (zero-24) [reply components/com_users/controllers/registration.php]:
```suggestion
```

File: components/com_users/controllers/registration.php [hunk 3]
```diff
@@ -63,7 +63,9 @@ public function activate()
 
 		if (!$userIdToActivate)
 		{
-			JError::raiseError(403, JText::_('COM_USERS_ACTIVATION_TOKEN_NOT_FOUND'));
+
```
Comment (zero-24) [inline components/com_users/controllers/registration.php]:
```suggestion
```

**Comment by zero-24** [comment]
@coolcat-creations The last changes should reflect the changes proposed by @Kubik-Rubik and @Quy

**Comment by zero-24** [comment]
Will await until the 4.0 version by @bembelimen has been merged and tested.

**Comment by bembelimen** [comment]
Done

**Comment by zero-24** [comment]
Merging your PR not merging the branch in here ;)

**Comment by coolcat-creations** [comment]
@zero-24 anything for me to change in the code? Or did you already do it?

**Comment by zero-24** [comment]
I did the intended changes. No additional changes needed for now we are awaiting  the decision from 4.0-dev

**Comment by Quy** [comment]
#35574 Merged

**Comment by zero-24** [comment]
Merging now. Thanks

**Pull request merged**